src/storage/src/hummock/compactor/shared_buffer_compact.rs

```diff
@@ -359,78 +347,24 @@ pub async fn merge_imms_in_memory(
         .first()
         .map(|((k, _), _)| k.clone())
         .unwrap_or_default();
-    let mut monotonic_tombstone_events = vec![];
-    let target_extended_user_key =
-        PointRange::from_user_key(UserKey::new(table_id, TableKey(pivot.as_ref())), false);
-    while del_iter.is_valid() && del_iter.key().le(&target_extended_user_key) {
-        let event_key = del_iter.key().to_vec();
-        del_iter.next().await?;
-        monotonic_tombstone_events.push(MonotonicDeleteEvent {
-            event_key,
-            new_epoch: del_iter.earliest_epoch(),
-        });
-    }
 
     let mut versions: Vec<(EpochWithGap, HummockValue<Bytes>)> = Vec::new();
 
-    let mut pivot_last_delete_epoch = HummockEpoch::MAX;
-
     for ((key, value), epoch) in items {
         assert!(key >= pivot, "key should be in ascending order");
         if key != pivot {
             merged_payload.push((pivot, versions));
             pivot = key;
-            pivot_last_delete_epoch = HummockEpoch::MAX;
             versions = vec![];
-            let target_extended_user_key =
-                PointRange::from_user_key(UserKey::new(table_id, TableKey(pivot.as_ref())), false);
-            while del_iter.is_valid() && del_iter.key().le(&target_extended_user_key) {
-                let event_key = del_iter.key().to_vec();
-                del_iter.next().await?;
-                monotonic_tombstone_events.push(MonotonicDeleteEvent {
-                    event_key,
-                    new_epoch: del_iter.earliest_epoch(),
-                });
-            }
-        }
-        let earliest_range_delete_which_can_see_key =
-            del_iter.earliest_delete_since(epoch.pure_epoch());
-        if value.is_delete() {
-            pivot_last_delete_epoch = epoch.pure_epoch();
-        } else if earliest_range_delete_which_can_see_key < pivot_last_delete_epoch {
```

Comment:
When there is no range delete in `del_iter`, the `earliest_delete_since` will return `u64::MAX`, and then we will never reach this branch.